src/Microsoft.Azure.SignalR.Management/Configuration/NegotiationResponseOptions.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Azure.SignalR.Management
+{
+    /// <summary>
+    /// Configuarable options for negotiation response options for redirecting client to SignalR service.
+    /// </summary>
+    public class NegotiationResponseOptions
+    {
+        /// <summary>
+        /// Gets or sets the flag whether the negotation response contains diagnostic client marker.
+        /// </summary>
+        public bool IsDiagnosticClient { get; set; } = false;
+
+        /// <summary>
+        /// The interval used by the server to timeout incoming handshake requests in SignalR service by clients. The default timeout is 15 seconds. The valid timeout is between 1 second and 30 seconds.
+        /// </summary>
+        public TimeSpan? HandshakeTimeout { get; set; } = null;
```

Comment:
Can `TimeSpan` type be resolved by the built-in `Bind` method? See [here ](https://github.com/Azure/azure-signalr/blob/a53d7b481aedcdc8fb5bbd93a04ad620968f8c9f/src/Microsoft.Azure.SignalR.Management/Configuration/ServiceManagerOptionsSetup.cs#L26). May need docs about configuring `TimeSpan` with string.

Reply:
See here: https://blog.ploeh.dk/2019/11/25/timespan-configuration-values-in-net-core/
built-in. doc can be in another pr for these options.